replicators/src/mysql_connector/connector.rs

```diff
@@ -80,8 +80,10 @@ impl MySqlBinlogConnector {
         info!(next_position = %self.next_position, "Starting binlog replication");
         let filename = self.next_position.binlog_file_name().to_string();
 
+        // TODO: we should require a re-snapshot if we are in the middle of a transaction and
+        //  the position has overflow u32.
         let cmd = mysql_common::packets::ComBinlogDump::new(self.server_id())
-            .with_pos(self.next_position.position)
+            .with_pos(self.next_position.position.try_into().unwrap())
```

Comment:
I assume this is the unwrap that'd fail if we try to start replicating from a point above `u32::MAX`. Could we turn this into an `expect` with a message so that we can know what's going on if we hit it?